text/0117-mobile-transactions-and-spans.md

```diff
@@ -0,0 +1,67 @@
+- Start Date: 2023-10-23
+- RFC Type: decision
+- RFC PR: https://github.com/getsentry/rfcs/pull/118
+- RFC Status: draft
+
+# Summary
+
+This RFC proposes to adapt the way transactions and spans are generated and sent on mobile platforms. In the recent light of [the performance API revamp](0101-revamping-the-sdk-performance-api.md) and the mobile starfish efforts we aim to provide a way of aggregating span data on a screen level.
+
+# Motivation
+
+For mobile starfish we want to be able to aggregate span data on a screen level. Metrics like TTID, TTFD, slow & frozen frames are especially useful if they can be grouped by screen, giving more actionable insights into the fast and slow parts of an app.
+
+# Background
+
+As of now the way transactions are created has several drawbacks:
+
+1. Missing root transaction: We automatically create transactions whenever an Activity (Android) or ViewController (iOS) is created, but they idle-timeout after 3s. This means we sometimes end up in situations where we can’t attach spans to a transaction as there's no running transaction.
+
+2. Multiple automatic transaction sources: We automatically start transactions for activities, user interactions and navigation events. These different sources are racing against each other as there can only be one transaction running (if a transaction is already running, a new one won’t be started), making it hard to predict the overall behavior.
+
+3. Automatic vs. manual transactions: Any user-created transaction potentially breaks the behavior of the existing automatic-transaction generation, since there can only be a single transaction on the scope.
+
+4. Transaction durations are sometimes pointless: As they’re based on idle transaction. Any relevant or even non-relevant child span can influence the transaction duration.
+
+# Options Considered
+
+### 1. Transactions as Carriers for Spans (preferred approach)
+
+The idea is to use transactions as carriers for measurements on mobile instead of trying to make the backend heavy concept work on mobile. There is no active "open" transaction, but we rather create transactions on-demand, whenever there is span data to send.
+In Sentry, we need pick apart these special transactions (that we shouldn’t consider as such anymore, maybe use `transaction->transaction_info->source` to identify carrier transaction) and only use their content, which are the spans we measured.
+
+TBD:
+* How can we ensure that spans are packaged up efficiently, ultimately creating not to little and not too many web requests.
```

Comment:
I think not a high prio thing to figure out, since eventually it will be Span Ingest.